inline.h

```diff
@@ -2677,19 +2676,24 @@ Perl_utf8_hop_forward(const U8 *s, SSize_t off, const U8 *end)
      * the bitops (especially ~) can create illegal UTF-8.
      * In other words: in Perl UTF-8 is not just for Unicode. */
 
-    assert(s <= end);
     assert(off >= 0);
 
+    if (UNLIKELY(s >= end)) {
+        GCC_DIAG_IGNORE(-Wcast-qual)
+        return (U8 *)end;
+        GCC_DIAG_RESTORE
+    }
+
```

Comment:
Shouldn't this just `assert(s <= end)`?

Reply:
My thinking on this was that where there is a possibility of accessing outside the buffer that the guards should be always enabled, and not just under DEBUGGING.  Also, it makes it behave sanely without crashing when called with a bad input value

Reply:
We could panic in both DEBUGGING and non-DEBUGGING builds, which is something we do in a variety of places.